src/sidebar/components/tag-editor.js

```diff
@@ -1,31 +1,38 @@
 import { createElement } from 'preact';
-import { useMemo, useRef, useState } from 'preact/hooks';
+import { useRef, useState } from 'preact/hooks';
 import propTypes from 'prop-types';
 
 import { withServices } from '../util/service-context';
 
+import Datalist from './datalist';
 import SvgIcon from './svg-icon';
+import useElementShouldClose from './hooks/use-element-should-close';
 
 // Global counter used to create a unique id for each instance of a TagEditor
-let datalistIdCounter = 0;
+let tagEditorIdCounter = 0;
 
 /**
  * Component to edit annotation's tags.
+ *
+ * Component accessibility is modeled after "Combobox with Listbox Popup Examples" found here:
+ * https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html
  */
 
 function TagEditor({ onEditTags, tags: tagsService, tagList }) {
   const inputEl = useRef(null);
-  const [showSuggestions, setShowSuggestions] = useState(false);
-  const [suggestionsFiltered, setSuggestionsFiltered] = useState([]); // For Chrome only
-  const [datalistId] = useState(() => {
-    ++datalistIdCounter;
-    return `tag-editor-datalist-${datalistIdCounter}`;
+  const [suggestions, setSuggestions] = useState([]);
+  const [activeItem, setActiveItem] = useState(-1); // -1 is unselected
+  const [isOpen, setOpen] = useState(false);
```

Comment:
I find `isOpen` slightly ambiguous in this context. `suggestionsListOpen` perhaps?